packages/nimble-components/src/table-column/table-column-text.ts

```diff
@@ -0,0 +1,81 @@
+import { attr, css, html, ViewTemplate } from '@microsoft/fast-element';
+import { DesignSystem } from '@microsoft/fast-foundation';
+import type { StringField, TableCellState } from '../table/types';
+import { TableColumn } from './table-column';
+import {
+    bodyFontColor,
+    bodyFont,
+    controlLabelFontColor
+} from '../theme-provider/design-tokens';
+
+type TableColumnTextCellData = StringField<'value'>;
+interface TableColumnTextColumnConfig {
+    placeholder: string;
+}
+
+/**
+ * The table column for displaying strings.
+ */
+export class TableColumnText extends TableColumn<
+TableColumnTextCellData,
+TableColumnTextColumnConfig
+> {
+    public cellStateDataFieldNames = ['value'] as const;
+
+    @attr({ attribute: 'field-name' })
+    public fieldName!: string;
+
+    @attr
+    public placeholder?: string;
+
+    public override readonly cellStyles = css`
+        .placeholder {
+            font: ${bodyFont};
+            color: ${controlLabelFontColor};
+            white-space: nowrap;
+        }
+
+        .text-value {
+            font: ${bodyFont};
+            color: ${bodyFontColor};
+            white-space: nowrap;
+        }
+    `;
+
+    public readonly cellTemplate: ViewTemplate<
+    TableCellState<TableColumnTextCellData, TableColumnTextColumnConfig>
+    > = html<
+        TableCellState<TableColumnTextCellData, TableColumnTextColumnConfig>
+        >`
+        <span
+            class="${x => (x.data.value !== null && x.data.value !== undefined
+        ? 'text-value'
+        : 'placeholder')}"
+        >
+            ${x => (x.data.value !== null && x.data.value !== undefined
+        ? x.data.value
+        : x.columnConfig.placeholder)}
+        </span>
+    `;
+
+    public getColumnConfig(): TableColumnTextColumnConfig {
+        return { placeholder: this.placeholder ?? '' };
+    }
+
+    public getRecordFieldNames(): string[] {
+        if (!this.fieldName) {
+            throw new Error('valueKey must be provided.');
+        }
+        return [this.fieldName];
+    }
+
+    public validateCellData(_: TableColumnTextCellData): void {}
+}
+
+const nimbleTableColumnText = TableColumnText.compose({
+    baseName: 'table-column-text'
+});
+
+DesignSystem.getOrCreate()
+    .withPrefix('nimble')
+    .register(nimbleTableColumnText());
```

Comment:
Two things I notice is that the table-column should have a stylesheet with `:host {display: none;}`
And that in the dom if I change the slotted header text then nothing happens. I think table-column needs a template with a slot that we can observe for changes.

Assuming those make sense, I think we need a default style and template registered for all of our table-columns. Maybe we should do a pattern like for icon with the registerIcon function that was exported from the same file as the Icon base class: https://github.com/ni/nimble/blob/main/packages/nimble-components/src/icon-base/index.ts#L27

Reply:
Done.

Reply:
Yeah, I intend to handle dynamic updates to the header content in a separate PR. It does not seem critical to address for the initial submission.